ext/opentelemetry-ext-docker-tests/tests/postgres/test_psycopg_functional.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+import unittest
+
+import psycopg2
+
+from opentelemetry import trace as trace_api
+from opentelemetry.ext.psycopg2 import trace_integration
+from opentelemetry.sdk.trace import Tracer, TracerProvider
+from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+from pytest import fixture
+
+POSTGRES_HOST = os.getenv("POSTGRESQL_HOST ", "localhost")
+POSTGRES_PORT = int(os.getenv("POSTGRESQL_PORT ", "5432"))
+POSTGRES_DB_NAME = os.getenv("POSTGRESQL_DB_NAME ", "opentelemetry-tests")
+POSTGRES_PASSWORD = os.getenv("POSTGRESQL_HOST ", "testpassword")
+POSTGRES_USER = os.getenv("POSTGRESQL_HOST ", "testuser")
+
+
+class TestFunctionalPsycopg(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._connection = None
+        cls._cursor = None
+        cls._tracer_provider = TracerProvider()
+        cls._tracer = Tracer(cls._tracer_provider, None)
+        cls._span_exporter = InMemorySpanExporter()
+        cls._span_processor = SimpleExportSpanProcessor(cls._span_exporter)
+        cls._tracer_provider.add_span_processor(cls._span_processor)
+        trace_integration(cls._tracer)
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls._cursor:
+            cls._cursor.close()
+        if cls._connection:
+            cls._connection.close()
+
+    @fixture(autouse=True)
+    def connect_to_db(self):
+        if self._connection is None:
+            # Try to connect to DB
+            for i in range(5):
+                try:
+                    self._connection = psycopg2.connect(
+                        dbname=POSTGRES_DB_NAME,
+                        user=POSTGRES_USER,
+                        password=POSTGRES_PASSWORD,
+                        host=POSTGRES_HOST,
+                        port=POSTGRES_PORT,
+                    )
+                    self._cursor = self._connection.cursor()
+                    break
+                except Exception as err:
+                    print(err)
+                time.sleep(5)
+            else:
+                raise Exception("Failed to connect to DB")
```

Comment:
instead of raising an exception that is unrelated to the real exception, how about, on the final loop, you just raise the exception?

Oftentimes printing an error doesn't give nearly enough context on what's going on. there's a lot lost without the traceback to see what code path is specifically failing.

Reply:
Real exception is being printed below so details will be available for each try and test error will be cleaner

Reply:
Hey hector, where is the real exception being printed? if you're referring to the block above, that only prints the exception message but doesn't print the traceback, which is valuable.